src/plugins/lua/bindings/lua_MQBindings.cpp

```diff
@@ -529,21 +529,42 @@ static void lua_pickle(sol::this_state L, std::string_view file_path, sol::table
 	if (ec)
 	{
 		LuaError("Failed to create directory for pickling %.*s with error: %s", file_path.size(), file_path.data(), ec.message());
-		return;
+		return false;
 	}
 
 	try
 	{
 		std::ofstream ofs(path, std::ios_base::out | std::ios_base::trunc);
 		ofs << fmt::to_string(buf);
 		ofs.close();
+		return true;
 	}
 	catch (std::exception e)
 	{
 		LuaError("Failed to write to file %.*s with error: %s", file_path.size(), file_path.data(), e.what());
+		return false;
 	}
 }
 
+static sol::table lua_unpickle(sol::this_state L, std::string_view file_path)
+{
+	std::filesystem::path fullPath = std::filesystem::path{ gPathConfig } / file_path;
+	std::string pathString = fullPath.string();
+
+	// Escape backslashes:
+	for (size_t pos = 0; (pos = pathString.find("\\", pos)) != std::string::npos; pos += 2) 
+	{
+		pathString.replace(pos, 1, "\\\\");
+	}
+
+	// Concatenate strings and run Lua code:
+	std::string luaCode =
+		"local fn, err = loadfile('" + pathString + "')\n"
+		"if not fn then error(err or '') end\n"
+		"return fn()";
```

Comment:
If you went this far you might as well just call this on the c++ side. All you're doing here is writing a script that does nothing but call one function, and we can do that better from c++.